pkg/config/network.go

```diff
@@ -58,3 +61,28 @@ func getClusterDNS(serviceCIDR string) (string, error) {
 
 	return dnsClusterIP.String(), nil
 }
+
+func (c *Config) EnsureNetworksDontOverlap() error {
+	linkList, err := netlink.LinkList()
+	if err != nil {
+		return fmt.Errorf("unable to load NIC list: %v", err)
+	}
+	for _, link := range linkList {
```

Comment:
This loop could use some logging calls to report what it's doing, which interfaces are OK, etc.